src/ImageSharp/Formats/Bmp/BMP Native Structures/BITMAPCOREHEADER.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Runtime.InteropServices;
+
+namespace SixLabors.ImageSharp.Formats.Bmp
+{
+    /// <summary>
+    /// This is the Microsoft Windows BMP v2 and IBM OS/2 BMP v1 (and later) DIB (Device Independent Bitmap) information header.
+    /// <para>Supported since Windows 2.0, Windows CE 2.0 and OS/2 1.0.</para>
+    /// <para>Implemented on Microsoft Windows BMP v2 and IBM OS/2 BMP v1 format.</para>
+    /// <seealso href="https://msdn.microsoft.com/en-us/library/dd183372(v=vs.85).aspx">See this MSDN link for more information.</seealso>
+    /// </summary>
+    /// <remarks>
+    /// Make shore that <c>sizeof(BITMAPCOREHEADER)</c> returns the size of 12 bytes and is byte aligned.
+    /// All structure fields are stored little-endian on the file.
+    /// <para>The colors (<seealso cref="RGBTRIPLE"/>) in the palette table should appear in order of importance
+    /// and must follow this structure.</para>
+    /// <para>Each scan line must be zero-padded to end on a DWORD (4 bytes) boundary.</para>
+    /// </remarks>
+    [StructLayout(LayoutKind.Sequential, Pack = 1, Size = 12)]
+    internal struct BITMAPCOREHEADER
```

Comment:
Can we use the same casing style here please as the rest of the solution. `BitmapCoreHeader`, Also rename the folder so there are no spaces.

Reply:
Yes, no problem.

I made the casing/naming more close to the SDKs for better interop between (as domentation of the BMP format) frameworks/implementations.

I will implement as the other decoder/encoders, using the class field read/write. So most of these structures will not be used, since I will avoid read/write the structures once and them swap the field for the correct CPU ending, if not on a little-ending platform (as more traditional used).

Reply:
Would you prefer that I add BMP images I use on testing as a link on a text file or the imges them selfs on the repository?

The major change to your code I made to your code until now was on "ImageSharp/src/ImageSharp/Formats/Bmp/BmpBitsPerPixel.cs", you have implemented it as "Bytes" insted of "bits", and only contemplated 1, 3 and 4 Bytes per pixel (8, 24 and 32 bits) and "ImageSharp/src/ImageSharpWould you prefer that I add BMP images I use on testing as a link on a text file or the images them self’s on the repository?

The major change to your code I made to your code until now was on "ImageSharp/src/ImageSharp/Formats/Bmp/BmpBitsPerPixel.cs", you have implemented it as "Bytes" insted of "bits", and only contemplated 1, 3 and 4 Bytes per pixel (8, 24 and 32 bits) and "ImageSharp/src/ImageSharp/Formats/Bmp/BmpFileHeader.cs" that implemted the 2 reserved WORDs (in the microsoft documentation) as one DWORD. All code dependent on these changes were change according.

All the native structures (as specified/documented by Microsoft and IBM) are implemented and documented on the “BMP Native Structures”. They will not be probably used in order to comply with the implementation that you have already used.

All de enumerations used by the specifications/documentation used by Microsoft and IBM are implemented and documented on the base project.  Note: “BmpBitsPerPixel.cs” enumeration has been change in a way that is incompatible with yours original version.

To easy the development, I created the "ImageSharp Picture Viewer" project. This gona be useful for testing RLE decompression (I think compression these days is a waste of time, because if compression in needed, PNG, JPEG or GIF, is a better format).
In this week I will implement the logic to decode/load all versions of file format and all the paletted files (as some people are requiring). Them I will implement the RLE and bitfields…
/Formats/Bmp/BmpFileHeader.cs" that implemted the 2 reserved WORDs (in the microsoft documentation) as one DWORD. All code dependent on these chanches were change acording.

All the native structures as spcified/documente by Microsoft and IBM.

In this week I will

Reply:
@caetanator we have [these unit tests](https://github.com/SixLabors/ImageSharp/blob/master/tests/ImageSharp.Tests/Formats/Bmp/BmpDecoderTests.cs) to test the BMP decoder. We need to add all test images covering the new headers to our test image suite to `tests/Images/Input/*".

Unfortunately we need to make sure they are not too big before pushing them. Can you share the links to your test images so I can have a look?